vscode/src/debugger/activate.ts

```diff
@@ -42,60 +42,50 @@ function registerCommands(context: vscode.ExtensionContext) {
-          // resolver fill it in with the currently open editor's URI.
-          // This will also let us correctly handle untitled files
-          // where the save prompt pops up before the debugger is launched,
-          // potentially causing the active editor URI to change if
-          // the file is saved with a different name.
-          vscode.debug.startDebugging(
-            undefined,
-            {
-              type: "qsharp",
-              name: "Run Q# File",
-              request: "launch",
-              shots: 1,
-              stopOnEntry: false,
-            },
-            { noDebug: true },
-          );
-        }
-      },
+      (resource: vscode.Uri) => startDebugging(resource, { noDebug: true }),
     ),
     vscode.commands.registerCommand(
       `${qsharpExtensionId}.debugEditorContents`,
-      (resource: vscode.Uri) => {
-        let targetResource = resource;
-        if (!targetResource && vscode.window.activeTextEditor) {
-          targetResource = vscode.window.activeTextEditor.document.uri;
-        }
-
-        if (targetResource) {
-          // We'll omit config.program and let the configuration
-          // resolver fill it in with the currently open editor's URI.
-          // This will also let us correctly handle untitled files
-          // where the save prompt pops up before the debugger is launched,
-          // potentially causing the active editor URI to change if
-          // the file is saved with a different name.
-          vscode.debug.startDebugging(undefined, {
-            type: "qsharp",
-            name: "Debug Q# File",
-            request: "launch",
-            shots: 1,
-            stopOnEntry: true,
-            noDebug: false,
-          });
-        }
-      },
+      (resource: vscode.Uri) => startDebugging(resource),
+    ),
+    vscode.commands.registerCommand(
+      `${qsharpExtensionId}.debugEditorContentsWithCircuit`,
+      (resource: vscode.Uri) =>
+        startDebugging(resource, {}, { showCircuit: true }),
     ),
   );
+
+  function startDebugging(
+    resource: vscode.Uri,
+    options?: vscode.DebugSessionOptions,
+    additionalConfig?: { [key: string]: any },
+  ) {
+    let targetResource = resource;
+    if (!targetResource && vscode.window.activeTextEditor) {
+      targetResource = vscode.window.activeTextEditor.document.uri;
+    }
+
+    if (targetResource) {
+      // We'll omit config.program and let the configuration
+      // resolver fill it in with the currently open editor's URI.
+      // This will also let us correctly handle untitled files
+      // where the save prompt pops up before the debugger is launched,
+      // potentially causing the active editor URI to change if
+      // the file is saved with a different name.
+      vscode.debug.startDebugging(
+        undefined,
+        {
+          type: "qsharp",
+          name: "Debug Q# File",
+          request: "launch",
+          shots: 1,
+          stopOnEntry: true,
```

Comment:
Good catch - refactoring mistake. It "works" in practice because we don't even check this value if not debugging, but I'll pass the value explicitly to reduce ambiguity.